OCPBUGS#33927: Add long-lived token/secret step to OLM policy scoping (#82232)
openshift/openshift-docs: modules/olm-policy-scoping-operator-install.adoc

. Create a new namespace:
+

> .Example creation of a `Namespace` object

Reviewer: I'm not sure about _Example creation_. Maybe just me. 
```suggestion
.Example that creates a `Namespace` object
```

  name: scoped
EOF
----
====

> . Allocate permissions that you want the Operator(s) to be confined to. This involves creating a new service account, relevant role(s), and role binding(s) in the newly created, designated namespace:

Reviewer: [Per the ISG](https://www.ibm.com/docs/en/ibm-style?topic=format-lists#lead-in-wording): Use a colon at the end of the lead-in sentence unless another sentence intervenes between the lead-in sentence and the first list item. If a sentence intervenes, use a period to end both the lead-in sentence and the intervening sentence.

```suggestion
. Allocate permissions that you want the Operator(s) to be confined to. This involves creating a new service account, relevant role(s), and role binding(s) in the newly created, designated namespace.
```

Author: ISG says to use colons leading into substeps.

.. Create a service account by running the following command:
+

> .Example creation of a `ServiceAccount` object

Reviewer: If you changed the earlier example, this one too.

  namespace: scoped
EOF
----
====

.. Create a secret by running the following command:
+

> .Example creation of a long-lived API token `Secret` object

Reviewer: If you changed the earlier example, this one too.

[%collapsible]
====
[source,terminal]
----
$ cat <<EOF | oc create -f -
apiVersion: v1
kind: Secret
type: kubernetes.io/service-account-token <1>
metadata:
  name: scoped
  namespace: scoped
  annotations:
    kubernetes.io/service-account.name: scoped
EOF
----
<1> The secret must be a long-lived API token, which is used by the service account.
====

.. Create a role by running the following command.
+
[WARNING]
====

> In this example, the role grants the service account permissions to do anything in the designated namespace for simplicity. In a production environment, you should create a more fine-grained set of permissions. For more information, see "Fine-grained permissions".

Reviewer: I know this is existing text, but it isn't immediately clear what _for simplicity_ refers to.

 ```suggestion
In this example, the role grants the service account permissions to do anything in the designated namespace for demonstration purposes only. In a production environment, you should create a more fine-grained set of permissions. For more information, see "Fine-grained permissions".
```

====
+

> .Example creation of `Role` and `RoleBinding` objects

Reviewer: If you changed the earlier example, this one too.

  namespace: scoped
EOF
----
====

. Create an `OperatorGroup` object in the designated namespace by running the following command. This Operator group targets the designated namespace to ensure that its tenancy is confined to it. In addition, Operator groups allow a user to specify a service account.
+

> .Example creation of an `OperatorGroup` object

Reviewer: If you changed the earlier example, this one too.

  name: scoped
  namespace: scoped
spec:
  serviceAccountName: scoped <1>
  targetNamespaces:
  - scoped
EOF
----
<1> Specify the service account created in the previous step. Any Operator installed in the designated namespace is tied to this Operator group and therefore to the service account specified.
====

. Create a `Subscription` object in the designated namespace to install an Operator:
+

> .Example creation of `Subscription` object

Reviewer: If you changed the earlier example, this one too.
